arcgis-ios-sdk-samples/Layers/ArcGIS vector tiled layer (custom style)/VectorTileCustomStyleViewController.swift

```diff
@@ -17,40 +17,90 @@ import UIKit
+    // Array of the item IDs.
+    private let itemIDs = ["1349bfa0ed08485d8a92c442a3850b06",
                            "bd8ac41667014d98b933e97713ba8377",
                            "02f85ec376084c508b9c8e5a311724fa",
-                           "1bf0cc4a4380468fbbff107e100f65a5"]
-    private var shownItemID: String? {
-        return ((mapView.map?.basemap.baseLayers.firstObject as? AGSArcGISVectorTiledLayer)?.item as? AGSPortalItem)?.itemID
+                           "1bf0cc4a4380468fbbff107e100f65a5",
+                           "2056bf1b350244d69c78e4f84d1ba215"]
+    // The item ID of the shown layer.
+    var shownItemID: String?
+    // The job to export the item resource cache.
+    var exportVectorTilesJob: AGSExportVectorTilesJob?
+    // The vector tiled layer created by the local VTPK.
+    var offlineVectorTiledLayer: AGSArcGISVectorTiledLayer?
+    
+    func loadOfflineLayer() {
+        let temporaryURL: URL = {
+            // Get a suitable directory to place files.
+            let directoryURL = FileManager.default.temporaryDirectory
+            // Create a unique name for the item resource cache based on current timestamp.
+            let temporaryFileName = UUID().uuidString
+            // Create and return the full, unique URL.
+            return directoryURL.appendingPathComponent("\(temporaryFileName)")
+        }()
+        // Get the vector tiled layer URL.
+        let vectorTiledLayerURL = URL(string: "https://arcgisruntime.maps.arcgis.com/home/item.html?id=2056bf1b350244d69c78e4f84d1ba215")!
+        // Create a vector tile cache using the local vector tile package.
+        let vectorTileCache = AGSVectorTileCache(name: "PSCC_vector")
+        // Create a portal item with the URL.
+        let portalItem = AGSPortalItem(url: vectorTiledLayerURL)!
+        // Create a task to export the custom style resources.
+        let task = AGSExportVectorTilesTask(portalItem: portalItem)
+        // Get the AGSExportVectorTilesJob.
+        exportVectorTilesJob = task.exportStyleResourceCacheJob(withDownloadDirectory: temporaryURL)
+        // Start the job.
+        exportVectorTilesJob?.start(statusHandler: nil) { [weak self] (result, error) in
+            guard let self = self else { return }
+            if let result = result {
+                // Get the item resource cache from the result.
+                let itemresourceCahce = result.itemResourceCache
+                // Create a vector tiled layer with the vector tiled cache and the item resource cache.
+                self.offlineVectorTiledLayer = AGSArcGISVectorTiledLayer(vectorTileCache: vectorTileCache, itemResourceCache: itemresourceCahce)
+                self.changeStyleBarItem.isEnabled = true
+            } else if let error = error {
+                // Handle errors.
+                self.presentAlert(error: error)
+            }
+        }
+        try? FileManager.default.removeItem(at: temporaryURL)
```

Comment:
This deletes the temporary directory before the job has even started. This should go inside of the completion handler.

Reply:
I tried to add it inside the completion handler but then the layer didn't appear at all. I thought it was possibly deleting the directory before the job finished. Is there an additional step I should do to include the deletion in the completion handler?